src/Microsoft.AspNetCore.SignalR.Core/HubOptions.cs

```diff
@@ -2,27 +2,16 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.AspNetCore.SignalR
 {
     public class HubOptions
     {
-        /// <summary>
-        /// The default keep-alive interval. This is set to exactly half of the default client timeout window,
-        /// to ensure a ping can arrive in time to satisfy the client timeout.
-        /// </summary>
-        public static readonly TimeSpan DefaultKeepAliveInterval = TimeSpan.FromSeconds(15);
+        public TimeSpan? NegotiateTimeout { get; set; } = null;
 
-        public TimeSpan NegotiateTimeout { get; set; } = TimeSpan.FromSeconds(5);
+        public TimeSpan? KeepAliveInterval { get; set; } = null;
 
-        /// <summary>
-        /// The interval at which keep-alive messages should be sent. The default interval
-        /// is 15 seconds.
-        /// </summary>
-        /// <remarks>
-        /// This interval is not used by the Long Polling transport as it has inherent keep-alive
-        /// functionality because of the polling mechanism. 
-        /// </remarks>
-        public TimeSpan KeepAliveInterval { get; set; } = DefaultKeepAliveInterval;
+        public List<string> SupportedProtocols { get; set; } = null;
```

Comment:
Make this `IList<string>`

We're saying `null` is "any protocol" right? Let's throw a doc comment on this explaining it as well.